nornir/core/inventory.py

```diff
@@ -438,3 +438,19 @@ def children_of_group(self, group: Union[str, Group]) -> Set[Host]:
             if host.has_parent_group(group):
                 hosts.add(host)
         return hosts
+
+    def add_host(self, name: str, defaults: Optional[Defaults] = None,
+                **kwargs) -> None:
+        """
+        Add a host to the inventory after initialization
+        """
+        host = {name: Host(name, defaults=self.defaults, **kwargs)}
```

Comment:
This is a really good start, however, in order to properly deserialize connection options you should use this method:

https://github.com/nornir-automation/nornir/blob/develop/nornir/core/deserializer/inventory.py#L78

So your code will be something like `host = {name: InventoryElement.deserialize_host(name=name, defaults=self.defaults, **kwargs)}`